src/irmin-pack/layered_store.ml

```diff
@@ -161,7 +161,7 @@ struct
   let find t k =
     let current = current_upper t in
     Log.debug (fun l -> l "find in %s" (log_current_upper t));
-    U.find current k >>= function
+    match U.unsafe_find ~check_integrity:true current k with
```

Comment:
this might not be necessary once #1153 is merged -- I've just seen memory pressure due to that bind